src/controller/java/src/matter/controller/model/States.kt

```diff
@@ -22,15 +22,61 @@ package matter.controller.model
  *
  * @param endpoints A mapping of endpoint IDs with the associated cluster data.
  */
-data class NodeState(val endpoints: Map<Int, EndpointState>)
+data class NodeState(val endpoints: MutableMap<Int, EndpointState> = mutableMapOf()) {
```

Comment:
You have modified NodeState to be more than a simple data container. It has mutable state (endpoints is a MutableMap) and provides behavior (methods like addAttribute, addEvent, and setDataVersion), so there is no point to use  data modifier for this class

Reply:
I will remove data modifier.